source/fundamentals/connection/mongoclientsettings.txt

```diff
@@ -0,0 +1,367 @@
+   <direct-connection-option>` parameter you can specify in your connection
+   URI.
+
+.. _mcs-socket-settings:
+
+Socket Settings
+---------------
+
+Chain the `applyToSocketSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToSocketSettings(com.mongodb.Block)>`__
+method to modify the driver's behavior when connecting and communicating
+with your MongoDB server.
+
+The following table describes all the methods you can chain to your
+settings to modify the driver's behavior:
+
+.. list-table::
+   :widths: 40 60
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``applyConnectionString​()``
+     - Uses the settings from a ``ConnectionString`` object.
+
+   * - ``applySettings()``
+     - Uses the socket settings specified in a ``SocketSettings`` object.
+
+   * - ``connectTimeout()``
+     - Sets the maximum time to connect to an available socket before throwing a timeout error.
+
+   * - ``readTimeout()``
+     - Sets the maximum time to read to an available socket before throwing a timeout error.
+
+   * - ``receiveBufferSize()``
+     - Sets the socket's buffer size when receiving.
+
+   * - ``sendBufferSize()``
+     - Sets the socket's buffer size when sending.
+
+Example
+~~~~~~~
+
+This example specifies the following driver behavior in a MongoDB socket:
+
+- To connect to an available socket within ``10 SECONDS``
+- To read from an available socket within ``15 SECONDS``
+
+.. literalinclude:: /includes/fundamentals/code-snippets/mcs.java
+   :start-after: begin SocketSettings
+   :end-before: end SocketSettings
+   :language: java
+   :emphasize-lines: 3-5
+   :dedent:
+
+.. _mcs-connectionpool-settings:
+
+Connection Pool Settings
+------------------------
+
+Chain the `applyToConnectionPoolSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToConnectionPoolSettings(com.mongodb.Block)>`__
+method to modify the way the driver manages its' connection pool.
+
+The following table describes all the methods you can chain to your
+settings to modify the driver's behavior:
+
+.. list-table::
+   :widths: 40 60
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``addConnectionPoolListener​()``
+     - Adds a listener for connection pool-related events.
+
+   * - ``applyConnectionString​()``
+     - Uses the settings from a ``ConnectionString`` object.
+
+   * - ``applySettings()``
+     - Uses the connection pool settings specified in a ``ConnectionPoolSettings`` object.
+
+   * - ``maintenanceFrequency()``
+     - Sets the frequency for running a maintenance job.
+
+   * - ``maintenanceInitialDelay()``
+     - Sets the time to wait before running the first maintenance job.
+
+   * - ``maxConnectionIdleTime()``
+     - Sets the amount of time a connection can be idle before it's closed.
+
+   * - ``maxConnectionLifeTime()``
+     - Sets the maximum time a pooled connection can live for.
+
+   * - ``maxSize()``
```

Comment:
Let's take this opportunity to clarify a point of confusion for many users by saying something like:

The maxSize setting applies to each server in the cluster to which the driver is connected.  So if the driver is connected to three mongos servers, there can be at most maxSize connections to _each_ mongos server.